<a href="https://colab.research.google.com/github/moulia-proma/ML_Practice/blob/main/LogisticRegression_MLP_XGBoost_to_predict_HeartDisease.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

importing libreries

In [1]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
import xgboost as xgb
from sklearn.neural_network import MLPClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score


addding data into dataframe

In [7]:
data = pd.read_csv('/content/Heart_Attack_Data_Set.csv')
data.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

checking rows and columns

In [8]:
data.shape

(303, 14)

counting the count of disease and no disease

In [9]:
data['target'].value_counts()

1    165
0    138
Name: target, dtype: int64

In [10]:
X=data.drop(columns='target',axis=1)
Y=data['target']



In [11]:
X

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  
0        0   0     1  
1        0   0     2  
2        2   0     2  
3        2   0     2  
4        2   0     2  
..     ...  ..   ...  
298      1   0     3  
299      1   0     3  
300      1   2     3  
301      1   1     3  
302      1   1     2  

[303 rows x 13 columns]

In [12]:
Y

0      1
1      1
2      1
3      1
4      1
      ..
298    0
299    0
300    0
301    0
302    0
Name: target, Length: 303, dtype: int64

In [13]:
X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size=0.1,stratify=Y,random_state=1)

In [14]:
print(X.shape,X_train.shape,X_test.shape)

(303, 13) (272, 13) (31, 13)


#LogisticRegression

In [15]:
model = LogisticRegression(solver='lbfgs', max_iter=1000)

In [16]:
model.fit(X_train,Y_train)

LogisticRegression(max_iter=1000)

In [17]:
ans=model.predict(X_train)

In [18]:
accuracy=accuracy_score(ans,Y_train)
precision=precision_score(ans,Y_train)
recall=recall_score(ans,Y_train)
f1=f1_score(ans,Y_train)

In [20]:
print("Training Accuracy is ",accuracy)
print("Training Precisiom is ",precision)
print("Training recall is ",recall)
print("Training F1 is ",f1)

Training Accuracy is  0.8419117647058824
Training Precisiom is  0.9121621621621622
Training recall is  0.8181818181818182
Training F1 is  0.8626198083067094


In [22]:
anss=model.predict(X_test)
accuracy_test_lr=accuracy_score(anss,Y_test)
precision_test_lr=precision_score(anss,Y_test)
recall_test_lr=recall_score(anss,Y_test)
f1_test_lr=f1_score(anss,Y_test)
print("Testing Accuracy is ",accuracy_test_lr)
print("Testing Precisiom is ",precision_test_lr)
print("Testing recall is ",recall_test_lr)
print("Testing F1 is ",f1)


Testing Accuracy is  0.9032258064516129
Testing Precisiom is  0.8823529411764706
Testing recall is  0.9375
Testing F1 is  0.8626198083067094


# LogisticRegression
#Result:
1.   Accuracy 90.32%
2.   Precision 88.23
3.Recall 93.75
4.F1 score 86.26




#XGBoost

In [24]:
model2=model = xgb.XGBClassifier()

In [25]:
model2.fit(X_train,Y_train)

XGBClassifier(base_score=None, booster=None, callbacks=None,
              colsample_bylevel=None, colsample_bynode=None,
              colsample_bytree=None, early_stopping_rounds=None,
              enable_categorical=False, eval_metric=None, feature_types=None,
              gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
              interaction_constraints=None, learning_rate=None, max_bin=None,
              max_cat_threshold=None, max_cat_to_onehot=None,
              max_delta_step=None, max_depth=None, max_leaves=None,
              min_child_weight=None, missing=nan, monotone_constraints=None,
              n_estimators=100, n_jobs=None, num_parallel_tree=None,
              predictor=None, random_state=None, ...)

In [31]:
acc = model2.predict(X_train)
accuracy_xgb=accuracy_score(acc,Y_train)
precision_xgb=precision_score(acc,Y_train)
recall_xgb=recall_score(acc,Y_train)
f1_xgb=f1_score(acc,Y_train)
print("Training Accuracy is ",accuracy_xgb)
print("Training Precisiom is ",precision_xgb)
print("Training recall is ",recall_xgb)
print("Training F1 is ",f1_xgb)

Training Accuracy is  1.0
Training Precisiom is  1.0
Training recall is  1.0
Training F1 is  1.0


In [38]:
acc2 = model2.predict(X_test)
accuracy_xgb_test=accuracy_score(acc2,Y_test)
precision_xgb_test=precision_score(acc2,Y_test)
recall_xgb_test=recall_score(acc2,Y_test)
f1_xgb_test=f1_score(acc2,Y_test)
print("Testing Accuracy is ",accuracy_xgb_test)
print("Testing Precisiom is ",precision_xgb_test)
print("Testing recall is ",recall_xgb_test)
print("Testing F1 is ",f1_xgb_test)

Testing Accuracy is  0.7419354838709677
Testing Precisiom is  0.6470588235294118
Testing recall is  0.8461538461538461
Testing F1 is  0.7333333333333334


#XGBoost
#Result :
1.   Accuracy is 74.19%
2.  Precision 64.70
3. Recall :84.61
4.F1:73.33



data.isnull()

In [ ]:
data.isnull().sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64

#MLPClassifier

In [53]:
model3 = MLPClassifier(hidden_layer_sizes=(100, 50), max_iter=1000,random_state=1)
model3.fit(X_train, Y_train)
y_pred = model3.predict(X_train)
acc_mlp  = accuracy_score(y_pred,Y_train)
precision_mlp=precision_score(y_pred,Y_train)
recall_mlp=recall_score(y_pred,Y_train)
f1_mlp=f1_score(y_pred,Y_train)
print("Testing Accuracy is ",acc_mlp)
print("Testing Precisiom is ",precision_mlp)
print("Testing recall is ",recall_mlp)
print("Testing F1 is ",f1_mlp)

Testing Accuracy is  0.8345588235294118
Testing Precisiom is  0.9324324324324325
Testing recall is  0.7976878612716763
Testing F1 is  0.8598130841121495


In [54]:
y_pred_t = model3.predict(X_test)
acc_mlp_t  = accuracy_score(y_pred_t,Y_test)
precision_mlp_t=precision_score(y_pred_t,Y_test)
recall_mlp_t=recall_score(y_pred_t,Y_test)
f1_mlp_t=f1_score(y_pred_t,Y_test)
print("Testing Accuracy is ",acc_mlp_t)
print("Testing Precisiom is ",precision_mlp_t)
print("Testing recall is ",recall_mlp_t)
print("Testing F1 is ",f1_mlp_t)

Testing Accuracy is  0.9032258064516129
Testing Precisiom is  0.9411764705882353
Testing recall is  0.8888888888888888
Testing F1 is  0.9142857142857143


#MLPClassifier
#Result:

1.   Acuuracy:90.32
2.   Precision:94.11
3.Recall:88.88
4.91.42

